### ISSUE TITLE
Some route names are generated with a prefix (prefix_except_default strategy)

### ISSUE THREAD

**Issue by Jassmin586** [nuxt-modules/i18n#1901]
### Environment

Nuxt project info:

------------------------------
- Operating System: `Linux`
- Node Version:     `v18.14.2`
- Nuxt Version:     `3.1.1`
- Nitro Version:    `2.1.0`
- Package Manager:  `yarn@1.22.19`
- Builder:          `vite`
- User Config:      `app`, `runtimeConfig`, `server`, `nitro`, `modules`, `i18n`, `googleFonts`, `typecript`, `css`, `components`, `hooks`, `vite`, `nuxtTypedRouter`, `extends`, `routeRules`
- Runtime Modules:  `@nuxtjs/tailwindcss@6.3.0`, `@pinia/nuxt@0.4.6`, `@vueuse/nuxt@9.12.0`, `nuxt-icons@3.1.0`, `@nuxtjs/google-fonts@3.0.0-1`, `vite-plugin-vue-type-imports/nuxt`, `nuxt-typed-router@3.0.2`, `./modules/wellms`, `./modules/device`, `@nuxtjs/i18n@8.0.0-beta.9`
- Build Modules:    `-`
------------------------------

### Reproduction

https://stackblitz.com/edit/github-wx3txy?file=nuxt.config.ts

### Describe the bug

I am using the `prefix_except_default` strategy. Now I have one language translations for url and it is default.
Most route names are generated correctly.
But with nested slug/id or with number in route names is a problem - add a prefix at the end of route name.

![2023-03-02_09-12](https://user-images.githubusercontent.com/19594765/222400890-0b0fa0ac-1574-4529-bead-08db2a2a4066.png)

**Comment by kazupon** [comment]
Thank you for your reporting!

I have tried to understand what is wrong with your report.
However, I could not do it.
Could you please share some more context? 🙏

**Comment by Jassmin586** [comment]
Yes of course. I was using a package called `nuxt-typed-router` which creates variables and types based on route names.
I want to keep naming consistency in the `prefix_except_default` strategy.

![image](https://user-images.githubusercontent.com/19594765/223244211-ca50f030-58f5-4136-bb58-c66ca1b05534.png)

**Comment by kazupon** [comment]
Thank you for your more details.
The order of route prefixes cannot be changed currently. The route name separator can be changed using the `routesNameSeparator` option.

**Comment by BobbieGoede** [comment]
It's unclear whether this is an issue with nuxt-i18n or the other modules. I don't think I can update the dependencies in the reproduction to see if the issue is still present in the latest release.

Closing this for now, if you're still experiencing this issue please provide an updated minimal reproduction 🙏
